modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/ConfigureByDublinCoreTermWOH.java

```diff
@@ -96,29 +100,48 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
 
     String configuredCatalog = StringUtils.trimToEmpty(currentOperation.getConfiguration(DCCATALOG_PROPERTY));
     String configuredDCTerm = StringUtils.trimToEmpty(currentOperation.getConfiguration(DCTERM_PROPERTY));
+    String configuredExternalCatalog = StringUtils.trimToEmpty(currentOperation.getConfiguration(EXTERNAL_CATALOG_PROPERTY));
     String configuredDefaultValue = StringUtils.trimToNull(currentOperation.getConfiguration(DEFAULT_VALUE_PROPERTY));
     String configuredMatchValue = StringUtils.trimToEmpty(currentOperation.getConfiguration(MATCH_VALUE_PROPERTY));
 
+    // Set Default
+    EName dcterm = null;
+    if (("").equals(configuredExternalCatalog)) {
+      configuredExternalCatalog = "dublincore";
+      dcterm = new EName(TERMS_NS_URI, configuredDCTerm);
+    }
     // Find Catalog
     Catalog[] catalogs = mediaPackage
-            .getCatalogs(new MediaPackageElementFlavor("dublincore", StringUtils.lowerCase(configuredCatalog)));
+            .getCatalogs(new MediaPackageElementFlavor(configuredExternalCatalog, StringUtils.lowerCase(configuredCatalog)));
 
     if (catalogs != null && catalogs.length > 0) {
       Boolean foundValue = false;
-      EName dcterm = new EName(TERMS_NS_URI, configuredDCTerm);
 
       // Find DCTerm
+      List<DublinCoreValue> values = new LinkedList<>();
       for (Catalog catalog : catalogs) {
         DublinCoreCatalog dc = DublinCoreUtil.loadDublinCore(workspace, catalog);
-        // Match Value
-        List<DublinCoreValue> values = dc.get(dcterm);
-        if (values.isEmpty()) {
-          // Use default
-          if (configuredDefaultValue != null) {
-            foundValue = configuredDefaultValue.equals(configuredMatchValue);
+          for (Map.Entry<EName, List<DublinCoreValue>> entry : dc.getValues().entrySet()) {
+            if (dcterm != null && dcterm.equals(entry)) {
```

Comment:
It's a bit weird that namespaces are used in case it's a dublincore catalog but that they are completely ignored otherwise. It would be less confusing to always (or never but I prefer always) specify them.

Reply:
The namespaces can be arbitrary selected by the user that introduces a new metadata catalogue. Where should I get these arbitrary namespaces from? Do we demand the users to provide them in another config file somewhere?

Removing the previously existing check for DC would be an option although.